src/sammich/plugins/contributors.py

```diff
@@ -46,36 +46,63 @@ def format_data(prs, issues, comments):
-
-    table = [
-        f"| {'User':<{max_name_length}} | PRs Merged | Issues Resolved | Comments |",
-        f"|{'-' * (max_name_length + 2)}|:{'-' * (max_prs_length + 2)}:|"
-        f":{'-' * (max_issues_length + 2)}:|:{'-' * (max_comments_length + 2)}:|",
-    ]
+    # Calculate column widths with minimum lengths
+    max_user_length = max(len(user) for user in user_data.keys())
+    max_prs_length = max(max(len(str(counts["prs"])) for counts in user_data.values()), 10)
+    max_issues_length = max(max(len(str(counts["issues"])) for counts in user_data.values()), 15)
+    max_comments_length = max(
+        max(len(str(counts["comments"])) for counts in user_data.values()), 9
+    )
+
+    # Formatting table header
+    header = (
+        f"{'User':<{max_user_length}}  "
+        f"{'PRs Merged':<{max_prs_length}}  "
+        f"{'Issues Resolved':<{max_issues_length}}  "
+        f"{'Comments':<{max_comments_length}}\n"
+    )
+    separator = (
+        f"{'-' * max_user_length}  "
+        f"{'-' * max_prs_length}  "
+        f"{'-' * max_issues_length}  "
+        f"{'-' * max_comments_length}\n"
+    )
+    rows = [header, separator]
+
+    # Formatting each user's data
     for user, counts in user_data.items():
-        table.append(
-            f"| {user:<{max_name_length}} | {counts['prs']:>{max_prs_length}} | "
-            f"{counts['issues']:>{max_issues_length}} | "
-            f"{counts['comments']:>{max_comments_length}} |"
+        row = (
+            f"{user:<{max_user_length}}  "
+            f"{counts['prs']:<{max_prs_length}}  "
+            f"{counts['issues']:<{max_issues_length}}  "
+            f"{counts['comments']:<{max_comments_length}}\n"
         )
+        rows.append(row)
+
+    table = "".join(rows)
 
-    return "\n".join(table)
+    return [
+        {
+            "type": "section",
+            "text": {"type": "mrkdwn", "text": f"*Contributor Data*\n```\n{table}\n```"},
+        }
+    ]
 
 
 class ContributorPlugin(MachineBasePlugin):
     """Contributor plugin"""
 
     @command("/contributors")
     async def contributors(self, command):
-        formatted_data = {}
-        try:
-            prs, issues, comments = fetch_github_data("OWASP-BLT", "BLT")
-            formatted_data = format_data(prs, issues, comments)
-        except requests.exceptions.JSONDecodeError:
-            pass
-
-        await command.say(formatted_data or "No data available")
+        channel_id = command._cmd_payload["channel_id"]
+        user = command._cmd_payload["user_id"]
+        prs, issues, comments = fetch_github_data("OWASP-BLT", "BLT")
+        formatted_data = format_data(prs, issues, comments)
+        if not formatted_data:
+            formatted_data = [
+                {"type": "section", "text": {"type": "mrkdwn", "text": "No data available"}}
+            ]
+
+        await self.web_client.chat_postEphemeral(
```

Comment:
You can use higher level [API](https://dondebonair.github.io/slack-machine/api/#machine.plugins.base.MachineBasePlugin.say) for communication (e.g. `self.say()`)

Reply:
.say() . not allowed to use in rich formatting like block.

Reply:
The docs say the opposite:

> Send a message to a channel using the WebAPI. Allows for rich formatting using blocks and/or attachments/

Could you provide more details supporting your statement?